File: t/VB/MedicalViewerDemo/UI/ImageAlignmentDialog.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Imports Leadtools
Imports Leadtools.MedicalViewer
Imports Leadtools.ImageProcessing.Core

Namespace MedicalViewerDemo
   Partial Public Class ImageAlignmentDialog : Inherits Form
      Private _form As MainForm
      Public Options As RegistrationOptions
      Public OldOptions As RegistrationOptions

      Public Sub New(ByVal form As MainForm)
         _form = form
         InitializeComponent()
      End Sub

      Private Sub _btnApply_Click(ByVal sender As Object, ByVal e As EventArgs) Handles _btnApply.Click
         _form.ApplyAlignment()
      End Sub

      Private Sub _btnReset_Click(ByVal sender As Object, ByVal e As EventArgs) Handles _btnReset.Click
         _form.ResetAlignment()
         EnableApplyButton(False)
      End Sub

      Private Sub _bntCancel_Click(ByVal sender As Object, ByVal e As EventArgs) Handles _bntCancel.Click
         Me.Close()
      End Sub

      Private Sub ImageAlignmentDialog_FormClosing(ByVal sender As Object, ByVal e As FormClosingEventArgs) Handles Me.FormClosing
         _form.FinishAlignment()
      End Sub

      Private Sub ImageAlignmentDialog_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
         _cbTransformation.SelectedIndex = 0
      End Sub

      Private Sub _cbTransformation_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles _cbTransformation.SelectedIndexChanged
         Dim type As Integer = _cbTransformation.SelectedIndex

         Select Case type
            Case 0
               Options = RegistrationOptions.Unknown
            Case 1
               Options = RegistrationOptions.XY
            Case 2
               Options = RegistrationOptions.RSXY
            Case 3
               Options = RegistrationOptions.Affine6
            Case 4
               Options = RegistrationOptions.Perspective
         End Select

         'if (_form.TemplateList.Count >= 1 && OldOptions != Options)
         '    EnableApplyButton(true);
         'else
         '    EnableApplyButton(false);
      End Sub

      Public Sub EnableApplyButton(ByVal Enable As Boolean)
         Me._btnApply.Enabled = Enable
      End Sub

   End Class
End Namespace

File: t/CS/MedicalViewerDemo/UI/ImageAlignmentDialog.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using Leadtools;
using Leadtools.MedicalViewer;
using Leadtools.ImageProcessing.Core;

namespace MedicalViewerDemo
{
    public partial class ImageAlignmentDialog : Form
    {
        private MainForm _form;
        public RegistrationOptions Options;
        public RegistrationOptions OldOptions;

        public ImageAlignmentDialog(MainForm form)
        {
            _form = form;
            InitializeComponent();
        }

        private void _btnApply_Click(object sender, EventArgs e)
        {
            _form.ApplyAlignment();
        }

        private void _btnReset_Click(object sender, EventArgs e)
        {
            _form.ResetAlignment();
            EnableApplyButton(false);
        }

        private void _bntCancel_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        private void ImageAlignmentDialog_FormClosing(object sender, FormClosingEventArgs e)
        {
            _form.FinishAlignment();
        }

        private void ImageAlignmentDialog_Load(object sender, EventArgs e)
        {
            _cbTransformation.SelectedIndex = 0;
        }

        private void _cbTransformation_SelectedIndexChanged(object sender, EventArgs e)
        {
            int type = _cbTransformation.SelectedIndex;

            switch (type)
            {
                case 0:
                    Options = RegistrationOptions.Unknown;
                    break;
                case 1:
                    Options = RegistrationOptions.XY;
                    break;
                case 2:
                    Options = RegistrationOptions.RSXY;
                    break;
                case 3:
                    Options = RegistrationOptions.Affine6;
                    break;
                case 4:
                    Options = RegistrationOptions.Perspective;
                    break;
            }

            //if (_form.TemplateList.Count >= 1 && OldOptions != Options)
            //    EnableApplyButton(true);
            //else
            //    EnableApplyButton(false);
        }

        public void EnableApplyButton(bool Enable)
        {
            this._btnApply.Enabled = Enable;
        }

    }
}
